pkg/common/common.go

```diff
@@ -45,3 +46,29 @@ func ReplaceStringsInFile(fileName string, replacements map[string]string) error
 
 	return nil
 }
+
+// ReplacePatternsInFile uses regexes to replace patterns with replacements in manifests during runtime.
+func ReplacePatternsInFile(fileName string, replacements map[string]string) error {
```

Comment:
I don't think we need two separate `func`s to handle strings or regular expressions. One should suffice. See example https://go.dev/play/p/-5A8-bkarvS

Also, instead of `regexp.MustCompile` which panics perhaps it's better use `regexp.Compile` and push the error down the call stack so that it can be managed appropriately at a higher level like in other cases.

Reply:
nice, I like this solution, working on implementing this now

Reply:
implemented here - consolidated both into one with a helper func https://github.com/maistra/opendatahub-operator/pull/14/commits/9613807ec3c807eed5cfceac7c15f63cfab1b946